vsphere_server_plugin/server.py

```diff
@@ -812,14 +812,24 @@ def _get_existing_server_details(ctx, server_client, server_obj):
 
 
 def get_vm_name(ctx, server, os_family):
-    # VM name may be at most 15 characters for Windows.
+    # If we've already set the name on this instance, use that.
+    vm_name = ctx.instance.runtime_properties.get('name')
+    if vm_name is not None:
+        return vm_name
+
+    # Gather up the details that we will use to select a name.
+    configured_name = server.get('name')
+    add_scale_suffix = server.get('add_scale_suffix', True)
+    instance_name, id_suffix = ctx.instance.id.rsplit('_', 1)
```

Comment:
move this line after `if configured_name is not None and not add_scale_suffix:` because results of operation unused if `add_scale_suffix:`.

It should be something like:
```python
configured_name = server.get('name')
add_scale_suffix = server.get('add_scale_suffix', True)
if configured_name is not None and not add_scale_suffix:
        return configured_name

instance_name, id_suffix = ctx.instance.id.rsplit('_', 1)
name_prefix = configured_name or instance_name
```

Reply:
done, thanks!